src/play_scraper.py

```diff
@@ -26,31 +30,49 @@ def __init__(self, base_url: str, storage_dir: Path):
         self._storage_dir = storage_dir
         self._storage_dir.mkdir(exist_ok=True, parents=True)
 
-    def get_icon(self, app_id: str, directory: Path = "") -> None:
-        """Downloads the app's icon from the corresponding web page.
+    def _build_app_page_url(self, app_id: str) -> str:
+        return self._base_url + app_id
 
-        :param app_id: the id of the app.
-        :param directory: icon storage subdirectory.
-        """
-        url = self._base_url + app_id
-        html = _get_html(url)
+    def _get_details_page(self, app_id: str) -> BeautifulSoup:
+        url = self._build_app_page_url(app_id)
+        return _get_html(url)
 
+    def _scrape_icon_url(self, html: BeautifulSoup) -> str:
         icon = html.find(class_=self._ICON_CLASS)
-        src = icon["src"]
+        if icon is None:
+            raise PlayScrapingError("Icon class not found in html.")
+        return icon["src"]
 
+    def _download_icon(self, app_id: str, src: str, directory: Path) -> None:
         location = self._storage_dir / directory
-        location.mkdir(exist_ok=True)
+        location.mkdir(exist_ok=True, parents=True)
 
         urlretrieve(src, location / f"icon_{app_id}")
 
+    def _scrape_category(self, html: BeautifulSoup) -> str:
+        category = html.find(itemprop=self._CATEGORY_ITEMPROP)
+        if category is None:
+            raise PlayScrapingError("Category itemprop not found in html.")
+        return category.get_text()
+
+    def get_icon(self, app_id: str, directory: Path = "") -> None:
+        """Scrapes the app icon URL from the app's Play Store details page,
+        downloads the corresponding app icon and saves it to the specified
+        subdirectory.
```

Comment:
Add an explicit comment that the app icon is stored at storage_dir/sub_dir/icon_{app_id}